pulumi/infra/fargate_service.py

```diff
@@ -0,0 +1,344 @@
+import json
+from typing import Mapping, Optional, Sequence, Tuple, Union
+
+import pulumi_aws as aws
+import pulumi_docker as docker
+from infra.cache import Cache
+from infra.config import DEPLOYMENT_NAME
+from infra.emitter import EventEmitter
+from infra.metric_forwarder import MetricForwarder
+from infra.network import Network
+from infra.policies import ECR_TOKEN_POLICY, attach_policy
+from infra.repository import Repository, registry_credentials
+from infra.service_queue import ServiceQueue
+
+import pulumi
+
+
+class FargateTaskRole(aws.iam.Role):
+    def __init__(
+        self, name: str, opts: Optional[pulumi.ResourceOptions] = None
+    ) -> None:
+        super().__init__(
+            f"{name}-task-role",
+            description=f"Fargate task role for {name}",
+            assume_role_policy=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Action": "sts:AssumeRole",
+                            "Principal": {"Service": "ecs-tasks.amazonaws.com"},
+                        }
+                    ],
+                }
+            ),
+            opts=opts,
+        )
+
+
+class FargateExecutionRole(aws.iam.Role):
+    def __init__(
+        self, name: str, opts: Optional[pulumi.ResourceOptions] = None
+    ) -> None:
+        super().__init__(
+            f"{name}-execution-role",
+            description=f"Fargate execution role for {name}",
+            assume_role_policy=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Action": "sts:AssumeRole",
+                            "Principal": {"Service": "ecs-tasks.amazonaws.com"},
+                        }
+                    ],
+                }
+            ),
+            opts=opts,
+        )
+
+
+class AWSFargateService(pulumi.ComponentResource):
```

Comment:
Can we maybe make this `_AWSFargateService` just so nobody tries to consume  it instead of the subclass?